src/Helpers/Consign/index.ts

```diff
@@ -0,0 +1,113 @@
+  props: ExcludeDefined<FocusedOnSearchInput>,
+): FocusedOnSearchInput => {
+  return {
+    action: ActionType.focusedOnSearchInput,
+    context_module: ContextModule.priceEstimate,
+    context_owner_id: props.context_owner_id,
+    context_owner_slug: props.context_owner_slug,
+    context_owner_type: OwnerType.consign,
+  }
+}
+
+export const selectedItemFromSearch = (
+  props: ExcludeDefined<SelectedItemFromSearch>,
+): SelectedItemFromSearch => {
+  return {
+    action: ActionType.selectedItemFromSearch,
+    context_module: ContextModule.priceEstimate,
+    context_owner_id: props.context_owner_id,
+    context_owner_slug: props.context_owner_slug,
+    context_owner_type: OwnerType.consign,
+    destination_owner_id: props.destination_owner_id,
+    destination_owner_slug: props.destination_owner_slug,
+    destination_owner_type: props.destination_owner_type,
+    owner_id: props.owner_id,
+    owner_slug: props.owner_slug,
+    owner_type: props.owner_type,
+    query: props.query,
+  }
+}
+
+export const searchedWithNoResults = (
+  props: ExcludeDefined<SearchedWithNoResults>,
+): SearchedWithNoResults => {
+  return {
+    action: ActionType.searchedWithNoResults,
+    context_module: ContextModule.priceEstimate,
+    context_owner_id: props.context_owner_id,
+    context_owner_slug: props.context_owner_slug,
+    context_owner_type: OwnerType.consign,
+    destination_owner_type: props.destination_owner_type,
+    query: props.query,
+  }
+}
+
+export const clickedArtworkGroup = (
+  props: ExcludeDefined<ClickedArtworkGroup>,
+): ClickedArtworkGroup => {
+  return {
+    action: ActionType.clickedArtworkGroup,
+    context_module: ContextModule.artworkRecentlySoldGrid,
+    context_page_owner_type: OwnerType.consign,
+    destination_page_owner_id: props.destination_page_owner_id,
+    destination_page_owner_slug: props.destination_page_owner_slug,
+    destination_page_owner_type: props.destination_page_owner_type,
+    type: "thumbnail",
+  }
+}
+
+export const clickedArticleGroup = (
+  props: ExcludeDefined<ClickedArticleGroup>,
+): ClickedArticleGroup => {
+  return {
+    action: ActionType.clickedArticleGroup,
+    context_module: ContextModule.relatedArticles,
+    context_page_owner_type: OwnerType.consign,
+    destination_page_owner_id: props.destination_page_owner_id,
+    destination_page_owner_slug: props.destination_page_owner_slug,
+    destination_page_owner_type: props.destination_page_owner_type,
+    type: "thumbnail",
+  }
+}
+
+export const clickedAppDownload = (
+  props: ExcludeDefined<ClickedAppDownload>,
+): ClickedAppDownload => {
+  return {
+    action: ActionType.clickedAppDownload,
+    context_module: ContextModule.sellFooter,
+    context_page_owner_id: props.context_page_owner_id,
+    context_page_owner_slug: props.context_page_owner_slug,
+    context_page_owner_type: OwnerType.consign,
+    destination_path: props.destination_path,
+    subject: "Download the app",
```

Comment:
Should we add a `Subject` enum?

Reply:
Usually `subject` is a free-text string pulled in from the button